### ISSUE TITLE
Debian Jessie x64

### ISSUE THREAD

**Issue by sheykh-sleky** [trakt/Plex-Trakt-Scrobbler#286]
[com.plexapp.plugins.trakttv.log.zip](https://github.com/trakt/Plex-Trakt-Scrobbler/files/158946/com.plexapp.plugins.trakttv.log.zip)

I did post on plex.tv forums, but I turned up logging and tried another sync.  Uploading that log here.  More than happy to provide any further information.  Gracias!!

**Comment by sheykh-sleky** [comment]
Well, checking the history now, I show some updates.  I kicked something...  I still get failure's when manually running it.  Might there be a good way to test communication vs just stabbing in the dark?  I removed the attachment on the forum in favor of this platform.  Thanks again!

**Comment by sheykh-sleky** [comment]
Seems like it isn't updating my collection, when I add it locally it isn't being updated on trakt.  Manually set a few shows to watched, they aren't updating either.  Might be that the site takes more than just me syncing up before it publishes those changes?

**Comment by sheykh-sleky** [comment]
[com.plexapp.plugins.trakttv.log.zip](https://github.com/trakt/Plex-Trakt-Scrobbler/files/159406/com.plexapp.plugins.trakttv.log.zip)
Seems I've got it syncing up my watchlist - now I just need to figure out why it isn't updating my collection again.  As so far, it has been user error!  :-)

**Comment by fuzeman** [comment]
Looks like this error is causing syncs to fail:
```
Exception raised in run(): ('imdb', 'tt1990314')
Traceback (most recent call last):
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/trakt_sync/cache/main.py", line 39, in refresh
    for result in source.refresh(username):
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/trakt_sync/cache/sources/sync.py", line 87, in refresh
    self.update_store((username, media, data), store)
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/trakt_sync/cache/sources/core/base.py", line 37, in update_store
    collection['store'].delete(collection_keys - set(current.keys()))
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/trakt_sync/cache/backends/stash_.py", line 19, in delete
    return self.stash.delete(keys)
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/stash/main.py", line 23, in delete
    return self.algorithm.delete(keys)
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/stash/algorithms/lru.py", line 100, in delete
    node = self.nodes.pop(key)
KeyError: ('imdb', 'tt1990314')
```

Will try have a fix available to test later today.

**Comment by fuzeman** [comment]
Try syncing with this version of the plugin: https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/issue/286.zip

Post another log file if your collection still isn't being updated correctly.

**Comment by sheykh-sleky** [comment]
MONEY!  That fix is giving me 100% across the board.  I will say, it kicked off a full scan.  And failed.  I kicked another off and it failed.  I had to run a push from TV and a push from Movies (the bulk of my libraries) before a full was happening each time.  It kept timing out due to "overload" and "too much sh*t to sync".  heh - I'll keep my eye on it, but it looks like this is working on my system.  Again, much thanks fuzeman!

Debian Jessie (headless) <>  x64 <>  PMS v0.9.15.6  <> 1.2k movies - 250 shows (7k episodes)

**Comment by fuzeman** [comment]
Glad that fix worked, will have it included in the next release.

>  It kept timing out due to "overload" and "too much sh*t to sync".

I might need to review how the plugin sends data to the Trakt.tv API, large requests should probably be split into a few smaller requests.

**Comment by sheykh-sleky** [comment]
Agreed.  Granted this is hopefully a one off for me, but I can see how a NEW library sync could be painful.  I don't think I have a large library either...  But I do find that every now and again adding a few seasons worth of a show could cause a timeout.  

Let me know if I can help/test in any way.  Checking the plugin console I'm seeing:

HTTPSConectionPool:  Max Retries exceeded with url: caused by ReadTimeOut (5 times, couple of different lists I have set-up)
Gateway Timeout "server timeout"

Love to solidify network or service level issues?  I wouldn't see any issues with port 443 on my network, but...  I've seen odd things in the lovely world of 'puters!

**Comment by fuzeman** [comment]
Fix released in [v1.0.1.2](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.0.1.2)
